prime-router/src/main/kotlin/history/azure/DeliveryFacade.kt

```diff
@@ -95,13 +95,27 @@ class DeliveryFacade(
      * @return Report details
      */
     fun findDetailedDeliveryHistory(
+        id: String,
```

Comment:
can this parameter be changed to `reportFileId` and then internally we use `parsedReportFileId`? At a glance from the function signature it's hard to tell what `id` is referring to. Update the javadoc comments as well.

Reply:
Yep this is because this can be iether a UUID or a Long.

Reply:
could split out into 2 functions and use the types and have a private shared implementation
```kotlin
fun findDetailedDeliveryHistory(
  reportId: UUID,
  deliveryId: Long,
)

fun findDetailedDeliveryHistory(
  actionId: Long,
  deliveryId: Long,
)
```

Reply:
So honestly it just pushes the same issue further up?  The root cause is the shared inheritance between the submission and delivery fetching, but I view that as out of scope for this work

Reply:
Looks like the design of this from the ground up has report ID and action ID mashed together so I agree it doesn't make sense to tackle it here.

Prying those types back apart in multiple places via UUID parsing exceptions sets off my code smell alarm bells though. An option for tech debt tickets down the line I guess 🤷

Reply:
I'll give myself a TODO to open a tech debt ticket for this.